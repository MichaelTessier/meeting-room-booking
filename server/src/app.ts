import express from 'express'

import bodyParser from 'body-parser'

import cors from 'cors'
import helmet from 'helmet'
import routes from './api/routes'
import { notFoundMiddleware } from './middlewares/not-found'
import { errorMiddleware } from './middlewares/error'
import { corsMiddleware } from './middlewares/cors'

const app = express()

app.use(express.static('public'))
app.use(bodyParser.json())

app.disable('x-powered-by')
// app.use(
//   cors({
//     origin: 'http://localhost:5173/',
//   }),
// )
app.use(corsMiddleware())
app.use(helmet())

// use body parser
// app.use(express.json())
app.use(express.urlencoded({ extended: true }))

app.use((req, res, next) => {
  res.setHeader('Access-Control-Allow-Origin', '*')
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST')
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type')
  res.setHeader('Access-Control-Allow-Credentials', 'true')
  next()
})

app.use('/api/v1', routes)
app.use(notFoundMiddleware)
app.use(errorMiddleware)

export default app
