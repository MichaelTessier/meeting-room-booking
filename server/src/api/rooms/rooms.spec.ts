import { describe, it, expect } from 'vitest'
import request from 'supertest'
import app from '@/app'

describe('API Rooms', () => {
  describe('GET /api/v1/rooms', () => {
    it('should return a formatted rooms', async () => {
      const response = await request(app).get('/api/v1/rooms')
      expect(response.status).toBe(200)
      expect(response.body).toMatchSnapshot()
    })
  })

  describe('GET /api/v1/rooms/:slug', () => {
    it('should return a formatted room', async () => {
      const response = await request(app).get('/api/v1/rooms/salle-1')
      expect(response.status).toBe(200)
      expect(response.body).toMatchSnapshot()
    })

    it('should return error if unknown slug', async () => {
      const response = await request(app).get('/api/v1/rooms/qsdqs')
      expect(response.status).toBe(404)
    })
  })
})
