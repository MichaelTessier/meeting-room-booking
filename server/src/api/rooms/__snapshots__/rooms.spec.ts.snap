// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`API Rooms > GET /api/v1/rooms > should return a formatted rooms 1`] = `
[
  {
    "capacity": 5,
    "createdAt": "2025-04-07T12:39:29.812Z",
    "description": "Espace d'expérimentation de test et d'idées",
    "equipments": [
      {
        "id": "1",
        "name": "TV",
        "type": "TELEVISION",
      },
      {
        "id": "2",
        "name": "Retro Projecteur",
        "type": "PROJECTOR",
      },
    ],
    "id": "1",
    "image": "undefined/images/campaign.jpg",
    "isAvailable": true,
    "name": "Lab Alpha",
    "slug": "lab-alpha",
    "updatedAt": "2025-04-08T17:31:39.489Z",
  },
  {
    "capacity": 10,
    "createdAt": "2025-04-07T12:39:55.384Z",
    "description": "Pour des prise de décisions rapides et précises",
    "equipments": [
      {
        "id": "2",
        "name": "Retro Projecteur",
        "type": "PROJECTOR",
      },
    ],
    "id": "2",
    "image": "undefined/images/cooper.jpg",
    "isAvailable": true,
    "name": "Quantum",
    "slug": "quantum",
    "updatedAt": "2025-04-07T13:33:37.184Z",
  },
  {
    "capacity": 11,
    "createdAt": "2025-04-07T14:15:55.733Z",
    "description": "Brainstorming et créativité",
    "equipments": [],
    "id": "3",
    "image": "undefined/images/hemant.jpg",
    "isAvailable": true,
    "name": "Helix",
    "slug": "helix",
    "updatedAt": "2025-04-09T16:45:19.025Z",
  },
  {
    "capacity": 10,
    "createdAt": "2025-04-09T16:45:34.419Z",
    "description": "Point de convergence de toutes les idées",
    "equipments": [
      {
        "id": "1",
        "name": "TV",
        "type": "TELEVISION",
      },
      {
        "id": "2",
        "name": "Retro Projecteur",
        "type": "PROJECTOR",
      },
    ],
    "id": "4",
    "image": "undefined/images/walls.jpg",
    "isAvailable": true,
    "name": "Nexus",
    "slug": "nexus",
    "updatedAt": "2025-04-09T16:45:34.419Z",
  },
  {
    "capacity": 26,
    "createdAt": "2025-04-09T16:45:49.096Z",
    "description": "Pour de la hauteur et de la perspective",
    "equipments": [
      {
        "id": "1",
        "name": "TV",
      },
      {
        "id": "2",
        "name": "Retro Projecteur",
        "type": "PROJECTOR",
      },
    ],
    "id": "5",
    "image": "undefined/images/wells.jpg",
    "isAvailable": true,
    "name": "Observatory",
    "slug": "observatory",
    "updatedAt": "2025-04-09T16:45:49.096Z",
  },
]
`;

exports[`API Rooms > GET /api/v1/rooms/:slug > should return a formatted room 1`] = `
{
  "capacity": 5,
  "createdAt": "2025-04-07T12:39:29.812Z",
  "description": "Espace d'expérimentation de test et d'idées",
  "equipments": [
    {
      "id": "1",
      "name": "TV",
      "type": "TELEVISION",
    },
    {
      "id": "2",
      "name": "Retro Projecteur",
      "type": "PROJECTOR",
    },
  ],
  "id": "1",
  "image": "undefined/images/campaign.jpg",
  "isAvailable": true,
  "name": "Lab Alpha",
  "slug": "lab-alpha",
  "updatedAt": "2025-04-08T17:31:39.489Z",
}
`;
